Cython/Utility/ModuleSetupCode.c

```diff
@@ -636,6 +636,23 @@ static CYTHON_INLINE void * PyThread_tss_get(Py_tss_t *key) {
 // PyThread_ReInitTLS() is a no-op
 #endif /* TSS (Thread Specific Storage) API */
 
+
+#if PY_MAJOR_VERSION < 3
+#if !CYTHON_COMPILING_IN_PYPY
+    // https://stackoverflow.com/a/25666624
+    static CYTHON_INLINE int PyGILState_Check(void) {
+        PyThreadState * tstate = _PyThreadState_Current;
+        return tstate && (tstate == PyGILState_GetThisThreadState());
+    }
+#else
+    static CYTHON_INLINE int PyGILState_Check(void) {
+        // completely unsatisfactory workaround, but it appears that it doesn't
+        // crash PyPy to try to release the GIL without holding the GIL.
+        return 0;
+    }
```

Comment:
Hmmmm... thinking about it more this is at least slightly dodgy, because it will end up with the GIL held even if it wasn't originally. It'll kind of work, and it doesn't seem to crash, but it's not really right.

@mattip do you know any better way of implementing this on PyPy2?

Reply:
(just to be clear since this confused me at first: PyPy supports this on python3.6+). I guess PyPy could backport the support for this function to PyPy2. The logic is pretty deep inside PyPy's implementation so I can't think of any other way to do this for PyPy.

Reply:
Added to PyPy2, this will be part of the next release (v7.3.6, probably in October).

Reply:
Thanks. I've looked at it more and decided that "return 0" is probably a fairly safe fallback - in this case it'll just result it in keeping hold of the GIL which is unlikely to be a disaster.

So I've updated the comment, added a PyPy version-check, and abused "deprecated" attributes to give a warning message for users to ignore. Which I think are all reasonable.